src/lib/components/segmentedControl/index.tsx

```diff
@@ -0,0 +1,102 @@
+import { useEffect, useRef, useState } from 'react';
+import styles from './style.module.scss';
+
+interface TitleWithSubtitle {
+  title: string;
+  subtitle: string;
+}
+
+const SegmentedControl = ({
+  className = '',
+  values,
+  selectedIndex,
+  onChange,
+}: {
+  className?: string;
+  values: Array<TitleWithSubtitle> | Array<string>;
+  selectedIndex: number;
+  onChange: (selectedIndex: number) => void;
+}) => {
+  const [selectedChipBackgroundWidthLeft, setSelectedChipBackgroundWidthLeft] =
+    useState<{
+      left: number;
+      width: number;
+    }>({ left: 0, width: 0 });
+  const chipContainer = useRef<any | null>(null);
```

Comment:
I don't know how strict we want to be with TS, but here we could use HTMLDivElement type to avoid using `any`.
That would mean an extra verification in the `useEffect` hook, but could potentially prevent type errors if this code is edited in the future.

Reply:
You're right, I've missed that part (I casted to any and do the proper typing later but I just forgot 😓 )
We could add a strict rule on the linter to avoid that.